core/util.go

```diff
@@ -144,6 +146,14 @@ func NewToken() string {
 	return RandomString(32)
 }
 
+const tokenFormat = "^[\\w-]{43}$"
+
+// LooksLikeAToken checks whether a string represents a 32-octet value in
+// the URL-safe base64 alphabet.
+func LooksLikeAToken(token string) bool {
+	return regexp.MustCompile(tokenFormat).MatchString(token)
```

Comment:
Since this gets run pretty often, it'd be a performance win to store the compiled regex and re-use it. That could be another issue/PR though.